src/components/AddressSearch/index.js

```diff
@@ -378,15 +378,17 @@ AddressSearch.propTypes = propTypes;
 AddressSearch.defaultProps = defaultProps;
 AddressSearch.displayName = 'AddressSearch';
 
+const AddressSearchWithRef = React.forwardRef((props, ref) => (
+    <AddressSearch
+        // eslint-disable-next-line react/jsx-props-no-spreading
+        {...props}
+        innerRef={ref}
+    />
+));
+
+AddressSearchWithRef.displayName = 'AddressSearchWithRef';
```

Comment:
The functions returned by `React.forwardRef` don't have an inherent name, therefore a `displayName` is required considering the new eslint rule added: `"react/display-name": 'error'`, otherwise it would throw lint error.

As soon as this is sorted out, I have @Pujan92's requested modifications in a stash, ready to be pushed, and will also sort the existing conflict.

I didn't want to mess anything up by pushing to the PR before the initial review.

Reply:
Can we merge this ?

cc @sobitneupane @roryabraham

Reply:
@ikevin127 I would like to know @roryabraham's thought before proceeding.

Reply:
Sorry for the long delay in replies here, I think it's fine that we set a displayName on the wrapper. This looks a bit weird but makes sense when you think about it. It will just show as `AddressSearchWithRef` in the React Dev Tools inspector instead of `Anonymous`.

Reply:
An alternative here might be to move the `React.forwardRef` to the initial function declaration instead of having a separate constant for it. Like this:

```diff
diff --git a/src/components/AddressSearch/index.js b/src/components/AddressSearch/index.js
index baa03ecc3a..d7d8cd5462 100644
--- a/src/components/AddressSearch/index.js
+++ b/src/components/AddressSearch/index.js
@@ -140,7 +140,7 @@ const defaultProps = {
 // Do not convert to class component! It's been tried before and presents more challenges than it's worth.
 // Relevant thread: https://expensify.slack.com/archives/C03TQ48KC/p1634088400387400
 // Reference: https://github.com/FaridSafi/react-native-google-places-autocomplete/issues/609#issuecomment-886133839
-function AddressSearch(props) {
+const AddressSearch = React.forwardRef((props, ref) => {
     const [displayListViewBorder, setDisplayListViewBorder] = useState(false);
     const [isTyping, setIsTyping] = useState(false);
     const [isFocused, setIsFocused] = useState(false);
@@ -401,19 +401,8 @@ function AddressSearch(props) {
                         }}
                         textInputProps={{
                             InputComp: TextInput,
-                            ref: (node) => {
-                                if (!props.innerRef) {
-                                    return;
-                                }
-
-                                if (_.isFunction(props.innerRef)) {
-                                    props.innerRef(node);
-                                    return;
-                                }
-
-                                // eslint-disable-next-line no-param-reassign
-                                props.innerRef.current = node;
-                            },
+                            // eslint-disable-next-line no-param-reassign
+                            ref: (node) => (ref.current = node),
                             label: props.label,
                             containerStyles: props.containerStyles,
                             errorText: props.errorText,
@@ -493,23 +482,10 @@ function AddressSearch(props) {
             {isFetchingCurrentLocation && <FullScreenLoadingIndicator />}
         </>
     );
-}
+});
 
 AddressSearch.propTypes = propTypes;
 AddressSearch.defaultProps = defaultProps;
 AddressSearch.displayName = 'AddressSearch';
 
-const AddressSearchWithRef = React.forwardRef((props, ref) => (
-    <AddressSearch
-        // eslint-disable-next-line react/jsx-props-no-spreading
-        {...props}
-        innerRef={ref}
-    />
-));
-
-AddressSearchWithRef.displayName = 'AddressSearchWithRef';
-
-export default compose(
-    withNetwork(),
-    withLocalize,
-)(AddressSearchWithRef);
+export default compose(withNetwork(), withLocalize)(AddressSearch);

```

Maybe we should discuss in slack